x/evm/types/chain_config.go

```diff
@@ -2,25 +2,171 @@ package types
+	ConstantinopleBlock sdk.Int `json:"constantinople_block" yaml:"constantinople_block"` // Constantinople switch block (< 0 no fork, 0 = already activated)
+	PetersburgBlock     sdk.Int `json:"petersburg_block" yaml:"petersburg_block"`         // Petersburg switch block (< 0 same as Constantinople)
+	IstanbulBlock       sdk.Int `json:"istanbul_block" yaml:"istanbul_block"`             // Istanbul switch block (< 0 no fork, 0 = already on istanbul)
+	MuirGlacierBlock    sdk.Int `json:"muir_glacier_block" yaml:"muir_glacier_block"`     // Eip-2384 (bomb delay) switch block (< 0 no fork, 0 = already activated)
+
+	YoloV1Block sdk.Int `json:"yoloV1_block" yaml:"yoloV1_block"` // YOLO v1: https://github.com/ethereum/EIPs/pull/2657 (Ephemeral testnet)
+	EWASMBlock  sdk.Int `json:"ewasm_block" yaml:"ewasm_block"`   // EWASM switch block (< 0 no fork, 0 = already activated)
+}
+
+// EthereumConfig returns an Ethereum ChainConfig for EVM state transitions.
+// All the negative or nil values are converted to nil
+func (cc ChainConfig) EthereumConfig(chainID *big.Int) *params.ChainConfig {
 	return &params.ChainConfig{
 		ChainID:             chainID,
-		HomesteadBlock:      big.NewInt(0),
-		DAOForkBlock:        big.NewInt(0),
+		HomesteadBlock:      getBlockValue(cc.HomesteadBlock),
+		DAOForkBlock:        getBlockValue(cc.DAOForkBlock),
+		DAOForkSupport:      cc.DAOForkSupport,
+		EIP150Block:         getBlockValue(cc.EIP150Block),
+		EIP150Hash:          common.HexToHash(cc.EIP150Hash),
+		EIP155Block:         getBlockValue(cc.EIP155Block),
+		EIP158Block:         getBlockValue(cc.EIP158Block),
+		ByzantiumBlock:      getBlockValue(cc.ByzantiumBlock),
+		ConstantinopleBlock: getBlockValue(cc.ConstantinopleBlock),
+		PetersburgBlock:     getBlockValue(cc.PetersburgBlock),
+		IstanbulBlock:       getBlockValue(cc.IstanbulBlock),
+		MuirGlacierBlock:    getBlockValue(cc.MuirGlacierBlock),
+		YoloV1Block:         getBlockValue(cc.YoloV1Block),
+		EWASMBlock:          getBlockValue(cc.EWASMBlock),
+	}
+}
+
+// String implements the fmt.Stringer interface
+func (cc ChainConfig) String() string {
+	out, _ := yaml.Marshal(cc)
+	return string(out)
+}
+
+// DefaultChainConfig returns default evm parameters. Th
+func DefaultChainConfig() ChainConfig {
+	return ChainConfig{
+		HomesteadBlock:      sdk.ZeroInt(),
+		DAOForkBlock:        sdk.ZeroInt(),
 		DAOForkSupport:      true,
-		EIP150Block:         big.NewInt(0),
-		EIP150Hash:          common.HexToHash("0x2086799aeebeae135c246c65021c82b4e15a2c451340993aacfd2751886514f0"),
-		EIP155Block:         big.NewInt(0),
-		EIP158Block:         big.NewInt(0),
-		ByzantiumBlock:      big.NewInt(0),
-		ConstantinopleBlock: big.NewInt(0),
-		PetersburgBlock:     big.NewInt(0),
+		EIP150Block:         sdk.ZeroInt(),
+		EIP150Hash:          "0x2086799aeebeae135c246c65021c82b4e15a2c451340993aacfd2751886514f0",
```

Comment:
I actually think this default hash is invalid should be populated on initialization

Reply:
@noot @araskachoi thoughts?